### PR TITLE
Fp 77

### PR DESCRIPTION
**Pull request by Archifer**

### CODE REVIEW

File: lib/pigeon.dart [hunk 1]
```diff
@@ -1382,9 +1382,6 @@ abstract class NativeCallFlutterApi {
   /// Called to open pin authentication screen.
```
Comment (MichielVrins) [inline lib/pigeon.dart:1382]: Can we check if we can rename this to contain generated in the name.

File: lib/constants/constants.dart [hunk 2]
```diff
@@ -193,9 +193,6 @@ class Constants {
   /// Event triggered by the finishRegistration needs to be responded
```
Comment (MichielVrins) [inline lib/constants/constants.dart:193]: Let's also remove the other unused constants.

File: example/lib/ow_broadcast_helper.dart [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+// ignore: import_of_legacy_library_into_null_safe
+import 'package:onegini_example/screens/pin_screen.dart';
+
+class OWBroadcastHelper {
+  static List<StreamSubscription<OWEvent>> initRegistrationListeners(BuildContext context) {
+    var broadCastController = Onegini.instance.userClient.owEventStreamController;
+
+    // Url Registration Related Events
+    StreamSubscription<OWEvent> handleRegisteredUrlSub = broadCastController.stream.where((event) => event is HandleRegisteredUrlEvent).listen((event) {
+      if (event is HandleRegisteredUrlEvent) {
```
[the remark this answers is not in the document]
Comment (Archifer) [reply example/lib/ow_broadcast_helper.dart]: decided that one broadcast stream is better to lower overhead

File: lib/constants/constants.dart (hunk 2 above)
Comment (Archifer) [reply lib/constants/constants.dart]: done; also removed unused models

File: lib/pigeon.dart (hunk 1 above)
Comment (Archifer) [reply lib/pigeon.dart:1385]: it is possible; i will do it after this branch is merged in another pr

File: lib/onegini_event_listener.dart [hunk 4]
```diff
@@ -1,160 +1,101 @@
...
+  void n2fCloseAuthOtp() {
+    broadcastEvent(CloseAuthOtpEvent());
   }
 
-  @override
-  void n2fEventPinNotAllowed(OWOneginiError error) {
-    pinNotAllowed(error);
+  /// Helper method
+  void broadcastEvent(OWEvent event) {
+    Onegini.instance.userClient.owEventStreamController.sink.add(event);
```
Comment (MichielVrins) [inline lib/onegini_event_listener.dart:99]: Can we make this method private and also add the streamController through the constructor?

File: lib/user_client.dart [hunk 5]
```diff
@@ -1,41 +1,35 @@
...
+import 'package:onegini/events/onewelcome_events.dart';
+import 'package:onegini/pigeon.dart';
 
 ///Сlass with basic methods available to the developer.
 class UserClient {
   final UserClientApi api;
   UserClient(this.api);
 
+  // ignore: close_sinks
+  final owEventStreamController = StreamController<OWEvent>.broadcast();
```
Comment (MichielVrins) [inline lib/user_client.dart:12]: Can we move this into the onegini.dart?

Comment (MichielVrins) [inline lib/user_client.dart:12]: This comment was added by me for convenience, can you investigate if we should actually close it?

File: lib/onegini_event_listener.dart (hunk 4 above)
Comment (Archifer) [reply lib/onegini_event_listener.dart:99]: done

File: lib/user_client.dart (hunk 5 above)
Comment (Archifer) [reply lib/user_client.dart:12]: done

Comment (Archifer) [reply lib/user_client.dart]: added finalizer to close the broadcaster before the onegini instance dies

File: example/lib/subscription_handlers/browser_registration_subscriptions.dart [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import 'dart:async';
+
+import 'package:onegini/events/browser_event.dart';
+import 'package:onegini/events/onewelcome_events.dart';
+import 'package:onegini/onegini.dart';
+import 'package:onegini/user_client.dart';
+import 'package:onegini_example/ow_broadcast_helper.dart';
+
+// Event Subscriptions related to the creation of Pin
+class BrowserRegistrationSubscriptions {
```
Comment (MichielVrins) [inline example/lib/subscription_handlers/browser_registration_subscriptions.dart:10]: Do these really need to be classes? Can't we just have them as normal functions and non-static?

File: example/lib/screens/login_screen.dart [hunk 7]
```diff
@@ -18,18 +21,34 @@ class LoginScreen extends StatefulWidget {
...
+    // Init subscriptipons for registration and authentication
+    this.registrationSubscriptions = OWBroadcastHelper.initRegistrationSubscriptions(context);
+    this.authenticationSubscriptions = OWBroadcastHelper.initAuthenticationSubscriptions(context);
+
     super.initState();
   }
 
+  @override
+  void dispose() {
+    print("disposing listeners");
```
Comment (MichielVrins) [inline example/lib/screens/login_screen.dart:39]: log not needed

**Pull request merged**